frontend/src/pages/Account/Account.tsx

```diff
@@ -7,41 +7,155 @@ import { FormLayout } from '../../shared/FormLayout';
+  Email = 'email',
+  Password = 'password',
+  NewPassword = 'newPassword',
 };
 
-
 export const Account: React.FC = () => {
-  const {
-    firstName,
-    lastName,
-    email,
-    password,
-    isSubscribed,
-  } = user;
+  const { user } = useAppSelector(state => state.auth);
+  const dispatch = useAppDispatch();
+  const [newValue, setNewValue] = useState<Partial<IRequestUserData>>({});
+  const [password, setPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const navigate = useNavigate();
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [fieldToChange, setFieldToChange] = useState<string>('');
+  const [isHideModal, setIsHideModal] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+
+  const handleGoBack = () => {
+    navigate(-1);
+  };
+
+  const handleOpenModal = (fieldName: string, event: MouseEvent) => {
+    event.stopPropagation();
+
+    setIsModalOpen(true);
+    setFieldToChange(fieldName);
+  };
+
+  const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const regex = new RegExp(/^[a-zA-Z0-9А-Яа-я \-\'\s]*$/);
+    const { name, value } = event.target;
+
+    if (!regex.test(value) && fieldToChange !== FieldNames.Email) {
+      return;
+    }
+
+    switch (name) {
+      case FieldNames.Password: {
+        setPassword(value);
+        break;
+      }
+      case FieldNames.NewPassword: {
+        setNewPassword(value);
+        break;
+      }
+      default:
+        setNewValue({[fieldToChange]: value});
+    }
+
+    setErrorMessage(null);
+  };
+
+  const closeModal = () => {
+    setIsHideModal(true);
+
+    const timer = setTimeout(() => {
+      setIsHideModal(false);
+      setIsModalOpen(false);
+      setFieldToChange('');
+      setPassword('');
+      setNewPassword('');
+      setNewValue({});
+      setErrorMessage(null);
+    }, 500);
+
+    return () => clearTimeout(timer);
+  };
+
+  const handleSubmit = async (event: FormEvent) => {
+    event.preventDefault();
+    const emailBody = {email: newValue.email?.toLowerCase(), password};
+    const passwordBody = {password, newPassword};
 
-  const fullName = `${firstName} ${lastName}`;
-  const [isEditName, setIsEditName] = useState<boolean>(false);
+    switch (fieldToChange) {
+      case FieldNames.Email: {
+        if (!newValue.email?.length) {
+          setErrorMessage('Enter Email');
+          return;
+        }
+
+        if (!password.length) {
+          setErrorMessage('Enter password');
+          return;
+        }
+        break;
+      }
+      case FieldNames.Password: {
+        if (!password.length) {
+          setErrorMessage('Enter password');
+          return;
+        }
+
+        if (!newPassword.length) {
+          setErrorMessage('Enter New password');
+          return;
+        }
+        break;
+      }
+    }
+
+    if (errorMessage?.length) {
+      return;
+    }
+
+    try {
+      switch (fieldToChange) {
```

Comment:
In general, there are too many conditions in this code.
This is beacaue you are trying to make a universal modal window.
You just have to create 4 separate modal windows, export them to separate components like `<UpdateUserFirstNameModal />` in features directory and implement all the logic inside the components.

Many conditions are bad practice because such code becomes difficult to support.